### PR TITLE
Add setup checks for sqlite driver & circles app

### PR DESCRIPTION
**Pull request by hweihwang**
### 📝 Summary

* Resolves: https://github.com/nextcloud/collectives/issues/1031

### 🏁 Checklist

- [x] Sqlite driver
- [x] Circles app

### CODE REVIEW

**Comment by juliushaertl** [comment]
The psalm failure is due to the fact that the collectives app supports multiple major nextcloud versions, so we'd need to safeguard the usage of those apis with a check if the method actually exists

File: lib/SetupChecks/CirclesAppIsEnableCheck.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		return 'app';
+	}
+
+	public function getName(): string {
+		return $this->l10n->t('Circles App Enabled');
+	}
+
+	public function run(): SetupResult {
+		if ($this->appManager->isEnabledForUser('circles')) {
+			return SetupResult::success($this->l10n->t('Circles App is enabled'));
```
Comment (juliushaertl) [inline lib/SetupChecks/CirclesAppIsEnableCheck.php:30]:
```suggestion
			return SetupResult::success($this->l10n->t('Teams app is enabled'));
```

We renamed the app to teams (sorry that wasn't reflected in the original ticket)

File: lib/SetupChecks/CirclesAppIsEnableCheck.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		}
+
+		return SetupResult::warning($this->l10n->t('Collectives app detected, but Circles app is not enabled. Please enable Circles app for enhanced collaboration with your team'));
```
Comment (juliushaertl) [inline lib/SetupChecks/CirclesAppIsEnableCheck.php:33]:
```suggestion
		return SetupResult::error($this->l10n->t('The teams app is not enabled, but is required for Collectives to work.'));
```

This is actually a hard requirement so we should make this more clear.

File: lib/SetupChecks/PDOSQLiteDriverIsEnableCheck.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		return $this->l10n->t('PDO SQLite driver');
+	}
+
+	public function run(): SetupResult {
+		if ($this->searchService->areDependenciesMet()) {
+			return SetupResult::success($this->l10n->t('PDO SQLite driver is enabled'));
+		}
+
+		return SetupResult::error(
+			$this->l10n->t('Collectives app detected, but PDO SQLite driver is missing. Please install for app functionality.')
```
Comment (juliushaertl) [inline lib/SetupChecks/PDOSQLiteDriverIsEnableCheck.php:34]:
```suggestion
			$this->l10n->t('Collectives app is enabled, but PDO SQLite driver is missing. Please install it to enable full text search of the page content.')
```

To be a bit clearer on the impact of not having this.

File: lib/SetupChecks/PDOSQLiteDriverIsEnableCheck.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		return 'database';
+	}
+
+	public function getName(): string {
+		return $this->l10n->t('PDO SQLite driver');
+	}
+
+	public function run(): SetupResult {
+		if ($this->searchService->areDependenciesMet()) {
+			return SetupResult::success($this->l10n->t('PDO SQLite driver is enabled'));
```
Comment (juliushaertl) [inline lib/SetupChecks/PDOSQLiteDriverIsEnableCheck.php:30]:
```suggestion
			return SetupResult::success($this->l10n->t('PDO SQLite driver is enabled, full text search of page content is available.'));
```

**Review by juliushaertl — COMMENTED** [review]
Just wording suggestions, generally looks good 👍

**Comment by hweihwang** [comment]
> Just wording suggestions, generally looks good 👍

Thank you :grin:

File: lib/AppInfo/Application.php [hunk 5]
```diff
@@ -100,21 +102,28 @@ public function register(IRegistrationContext $context): void {
...
-		if (\interface_exists(IAPIWidgetV2::class)) {
+		if (interface_exists(IAPIWidgetV2::class)) {
 			$context->registerDashboardWidget(RecentPagesWidget::class);
 		}
 
 		if (method_exists($context, 'registerTeamResourceProvider')) {
 			$context->registerTeamResourceProvider(CollectiveTeamResourceProvider::class);
 		}
+
+		if (interface_exists('OCP\SetupCheck\ISetupCheck') && method_exists($context, 'registerSetupCheck')) {
```
Comment (juliushaertl) [inline lib/AppInfo/Application.php:113]:
```suggestion
		if (interface_exists(\OCP\SetupCheck\ISetupCheck::class) && method_exists($context, 'registerSetupCheck')) {
```

We can use the class magic constant here

File: lib/SetupChecks/CirclesAppIsEnableCheck.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	) {
+	}
+
+
+	public function getCategory(): string {
+		return 'app';
+	}
+
+	public function getName(): string {
+		return $this->l10n->t('Circles App Enabled');
```
Comment (blizzz) [inline lib/SetupChecks/CirclesAppIsEnableCheck.php:25]:
```suggestion
		return $this->l10n->t('Teams App Enabled');
```

**Review by blizzz — COMMENTED** [review]
One more place for _Teams_ not _Cirles_

File: lib/AppInfo/composer.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+	"name": "vendor_name/AppInfo",
+	"description": "description",
+	"minimum-stability": "stable",
+	"license": "proprietary",
```
Comment (mejo-) [inline lib/AppInfo/composer.json:5]: Is this file needed?

Comment (hweihwang) [reply lib/AppInfo/composer.json]: Ah thank you, no need, let me remove it!

**Review by juliushaertl — APPROVED** [review]

**Review by blizzz — APPROVED** [review]
